docs/scope-and-objectives.md

```diff
@@ -0,0 +1,139 @@
+---
+title: Cluster API Scope and Objectives
+authors:
+  - "@vincepri"
+  - "@ncdc"
+  - "@detiber"
+reviewers:
+  - "@timothysc"
+  - "@justinsb"
+creation-date: 2019-04-08
+last-updated: 2019-04-08
+---
+
+# Cluster API Scope and Objectives
+
+This document is meant to be a living document refined over time. It is meant to serve as guardrails for what is in scope
+and out of scope for Cluster API. As new ideas or designs take shape over time, this document can and should be updated.
```

Comment:
This is probably a nitpicky comment, but one thing I always find difficult to explain is that cluster-api is not just an api, its an API and a set of controllers that are provider-independent. Maybe mention that it is both here?

What probably should also be mentioned (but not sure where) is that ppl are sort-of expected to use the controllers in this repo, because e.G. the current open cluster autoscaler PR relies on an implemention detail of the machineSet controller here (an annotation) that is not part of the API.

Reply:
Or maybe the annotation should be part of the spec.

Reply:
I agree that the project could use a better name. But, I don't want to get into implementation details in this document, so I would prefer not to mention controllers specifically.

I'm curious why you think people wouldn't use the Cluster API controllers? I fully expect that the outputs of the workstreams on extension mechanism, data model, and lifecycle management will result in an improved set of control logic that is always on, and any variations in execution are strictly from extension implementations (a.k.a. "providers").

Annotations that are defined and in use by a project are part of its API. They may not be standalone structs, but they are still part of the API and they must be treated as such, which includes adhering to compatibility guidelines and a deprecation policy. We can and should debate if these annotations in question should be annotations or if they should be spec fields, but regardless, they're still part of the API.

Reply:
I think it should be safe to call out that this repo/project is:
- An API definition for managing Kubernetes Clusters declaratively
- A reference implementation for the core logic
- An API definition for provider-specific extension mechanisms